docs/proposals/scheduling/multi-scheduling-group/README.md

```diff
@@ -0,0 +1,366 @@
+
+	// ReplicaScheduling represents the scheduling policy on dealing with the number of replicas
+	// when propagating resources that have replicas in spec (e.g. deployments, statefulsets) to member clusters.
+	// +optional
+	ReplicaScheduling *ReplicaSchedulingStrategy `json:"replicaScheduling,omitempty"`
+}
+// OrderedClusterAffinityTerm selects a set of clusters and indicates the order
+// that the Karmada scheduler should inspect.
+type OrderedClusterAffinityTerm struct {
+	ClusterAffinity `json:",inline"`
+
+	// Order of the cluster group that the Karmada scheduler should inspect with, in the range 1-100.
+	// +required
+	Order int32 `json:"order"`
+	// AffinityName is the name of the cluster group.
+	// +required
+    AffinityName string `json:"affinityName"`
+}
+```
+Each affinity term essentially is the current `ClusterAffinity` but has an
+ordinal number named `order` against which the scheduler evaluates each term individually.
+That means the scheduler always evaluates the first term(with `order: 1`), and turns
+to the next term if the term does not satisfy restrictions.
+
+The following configuration sample declares 3 affinity terms:
+```yaml
+apiVersion: policy.karmada.io/v1alpha1
+kind: PropagationPolicy
+metadata:
+  name: nginx
+spec:
+  resourceSelectors:
+    - apiVersion: apps/v1
+      kind: Deployment
+      name: nginx
+  placement:
+    orderedClusterAffinities:
+      - affinityName: dc-shanghai
+        order: 1
+        clusterNames:
+          - unavailable
+      - affinityName: dc-beijing
+        order: 2
+        clusterNames:
+          - member1
+      - affinityName: dc-hongkong
+        order: 3
+        clusterNames:
+          - member2
+```
+
+During the scheduling phase, the scheduler will first look at the affinity
+term with `order 1` and try to select a feasible cluster set from it.
+
+If no feasible cluster is found from the term, the scheduler will try against the
+next term which is the one with `order 2`. And once the scheduler successfully
+selected a feasible cluster set from this affinity term, the scheduler will not
+continue to look at the following terms.
+
+And, in case of cluster `member1` becomes unavailable, by leveraging the
+[Failover feature](https://karmada.io/docs/userguide/failover/failover-overview)
+the scheduler will start looking for alternatives `in current affinity term`, and
+if that fails, it will look at the term with `order 3`.
+
+**Note:** Each affinity term is completely independent, and the scheduler will
+only select one term at a time during scheduling. But it is allowed for the same
+cluster present in multiple affinity terms.
+
+#### ResourceBinding API change
+
+In order to track the affinity term that the scheduler last evaluated,
+the proposal proposes a new field named `SchedulerObservedAffinityIndex` to
+`.status` of `ResourceBinding`, so that the scheduler can continue the previous
+scheduling cycle.
```

Comment:
I'm sure about the effect of this field, could you give a case here about how scheduler leverage this field for a second scheduling?

Reply:
Re-schedule after an eviction or fail-over. The scheduler could continue the schedule from the last `affinity term`.

In addition, we also need to know if a re-schedule should be triggered in case of placement changes on PropagationPolicy.

Reply:
Does this mean that we need to distinguish the scenarios that trigger rescheduling?

Reply:
> In addition, we also need to know if a re-schedule should be triggered in case of placement changes on PropagationPolicy.

If the user deletes `ds-hongkong` in `orderedClusterAffinities`, will re-schedule?

Reply:
> If the user deletes ds-hongkong in orderedClusterAffinities, will re-schedule?

That depends on if the `ds-hongkong` is in use, that is I mentioned on https://github.com/karmada-io/karmada/issues/3153#issuecomment-1433007295.